include/options/business/domains.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\Options\Business;
+
+use WP_Error;
+use WP_Syntex\Polylang\Options\Primitive\Map;
+use WP_Syntex\Polylang\Options\Options;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class defining single associative array of domain as value and language slug as key option.
+ * /!\ Sanitization depends on `force_lang`: this option must be set AFTER `force_lang`.
+ *
+ * @since 3.7
+ *
+ * @phpstan-import-type SchemaType from \WP_Syntex\Polylang\Options\Abstract_Option
+ */
+class Domains extends Map {
+	/**
+	 * Returns the JSON schema part specific to this option.
+	 *
+	 * @since 3.7
+	 *
+	 * @return array Partial schema.
+	 *
+	 * @phpstan-return array{
+	 *     type: SchemaType,
+	 *     patternProperties: non-empty-array<non-empty-string, array{type: SchemaType, format: 'uri'}>,
+	 *     additionalProperties: bool
+	 * }
+	 */
+	protected function get_specific_schema(): array {
+		$map_schema                      = parent::get_specific_schema();
+		$map_schema['patternProperties'] = array(
+			'^[a-z_-]+$' => array( // Language slug as key.
+				'type'   => $this->type,
+				'format' => 'uri',
```

Comment:
This will require to override the constructor. Then, since `get_specific_schema()` is also overridden, extending `Map` is now far less interesting. So I'll do a commit to extend `Abstract_Option` directly instead of `Map`: no need to override the constructor anymore, and the type will be hard-coded.

Reply:
Done in [b902f15](https://github.com/polylang/polylang/pull/1451/commits/b902f1508bb1334fe3c9adb073a234bdc975123b).